src/common/components/course-card/index.tsx

```diff
@@ -0,0 +1,173 @@
+import React, { useEffect, useState } from "react";
+
+import DefaultCompanyImage from "~/assets/images/default-company-image.png";
+import { Icon } from "~/common/components/index";
+import { ScreenBreakpoints } from "~/common/constants/screen-breakpoints";
+import { IconName, RatingSize } from "~/common/enums";
+import { Course } from "~/common/types/courses/course.type";
+import RecentReview from "~/common/types/review/get-recent-review";
+
+import AvatarGroup from "../avatar-group";
+import { StarRating } from "../star-rating";
+import styles from "./styles.module.scss";
+
+type CourseCardProps = {
+	className?: string;
+	course: Course;
+};
+
+const CourseCard: React.FC<CourseCardProps> = ({ course }) => {
+	const CHAR_LIMIT_MOBILE = 107;
+	const CHAR_LIMIT_DESKTOP = 303;
+	const LAST_INDEX_OFFSET = 1;
+	const START_INDEX = 0;
+	const BASE_URL = import.meta.env.VITE_BASE_URL;
+	const AVATARS_COUNT = 4;
+
+	const [isExpanded, setIsExpanded] = useState(false);
+	const [isOneStar, setIsOneStar] = useState(false);
+	const [maxLength, setMaxLength] = useState(CHAR_LIMIT_DESKTOP);
+	const [avatars, setAvatars] = useState<string[]>([]);
+
+	const toggleDescription = () => {
+		setIsExpanded(!isExpanded);
+	};
+
+	useEffect(() => {
+		const handleResize = () => {
+			if (window.innerWidth < ScreenBreakpoints.TABLET) {
+				setMaxLength(CHAR_LIMIT_MOBILE);
+			} else {
+				setMaxLength(CHAR_LIMIT_DESKTOP);
+			}
+			setIsOneStar(window.innerWidth < ScreenBreakpoints.TABLET);
+		};
+
+		window.addEventListener("resize", handleResize);
+
+		handleResize();
+
+		return () => {
+			window.removeEventListener("resize", handleResize);
+		};
+	}, []);
+
+	useEffect(() => {
+		const fetchAvatars = async () => {
+			try {
+				const response = await fetch(
+					`${BASE_URL}/reviews/recent?count=${AVATARS_COUNT}`,
+				);
+				const data = await response.json();
+				const avatarsList = data.map(
+					(review: RecentReview) => review.author_avatar,
+				);
+				setAvatars(avatarsList);
+			} catch (error) {
+				// eslint-disable-next-line no-console
+				console.error("Failed to fetch avatars:", error);
+			}
+		};
+
+		fetchAvatars();
+	}, [BASE_URL]);
+
+	const companyLogo =
+		course.company_logo && course.company_logo !== "None"
+			? course.company_logo
+			: DefaultCompanyImage;
```

Comment:
i know it confused you, but it's not reallu default company avatar. we don't have default image. I think you can create div with some background color and show firts letter of company name is there isn't any image provided